File: nodejs/models/Order.js
// models/Order.js
const mongoose = require('mongoose');

const OrderSchema = new mongoose.Schema({
    tableId: { type: mongoose.Schema.Types.ObjectId, ref: 'Table' },
    items: [{
        productId: { type: mongoose.Schema.Types.ObjectId, ref: 'Product' },
        quantity: Number,
        price: Number
    }],
    totalAmount: Number,
    discount: { type: Number, default: 0 },
    staffName: String,
    status: { type: String, default: 'PAID' }, // PAID, CANCELLED
    createdAt: { type: Date, default: Date.now }
});

module.exports = mongoose.model('Order', OrderSchema);

File: android/java/com/example/posapp/domain/model/Order.kt
package com.example.posapp.domain.model

data class Order(
    val id: String,
    val items: List<OrderItem>,
    val total: Double,
    val isSynced: Boolean
)

data class OrderItem(
    val productId: String,
    val quantity: Int,
    val price: Double
)